Extensions/pokes.js

```diff
@@ -438,5 +442,37 @@ XKit.extensions.pokes = {
 
 		$("#xkit-extensions-panel-right").nanoScroller();
 		$("#xkit-extensions-panel-right").nanoScroller({ scroll: 'top' });
+	},
+
+	addSortID: function() {
+		var offset = parseInt(XKit.storage.get("pokes", "update_offset", 0));
+		if (isNaN(offset)) offset = 0;
+		var step_size = 1000;
+
+		if (offset !== -1) {
+			var pokes = JSON.parse(XKit.storage.get("pokes", "pokemon_storage", "[]"));
+			try {
+				XKit.extensions.pokes.fetch_pokedex(function(mdata) {
+					for (var i = offset; i < offset + step_size; i++) {
+						try {
+							pokes[i].sortid = mdata[pokes[i].id].sortid;
+						} catch (e) {
+							if (e instanceof ReferenceError) {
+								offset = -1; //This is good! We've finished the array and never have to do this ever again.
+							} else if (e instanceof TypeError) {
```

Comment:
If, for whatever reason, an entry is undefined, it will throw a TypeError. It's safe to skip those (just do nothing at all), so we can just ignore them completely.